httpc.go

```diff
@@ -460,7 +468,10 @@ func (r *Request) RunWithContext(ctx context.Context) error {
 		case <-ticker.C:
 		}
 
-		r.setBody(req)
+		err = r.setBody(req)
```

Comment:
Nitpick: Please use the `if err = r.setBody(req); err != nil {}` notation to keep the style consistent when returning in error case.